src/tests/unit/tests/common/promises/promise-factory.test.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { createDefaultPromiseFactory } from '../../../../../common/promises/promise-factory';
+
+// wrote the test following examples from https://jestjs.io/docs/en/tutorial-async
+describe('PromiseFactory', () => {
+    const testObject = createDefaultPromiseFactory();
+
+    describe('timeout', () => {
+        it('resolves', () => {
+            expect.assertions(1);
+
+            const actual = 'the result';
+            const resolving = Promise.resolve(actual);
+
+            const result = testObject.timeout(resolving, 10);
+
+            return expect(result).resolves.toEqual(actual);
```

Comment:
I was confused on first reading about how the test would avoid completing before the promise under test did; it took me a few readings of this to notice the `return` here and put together that the expectation itself was going to be treated as a promise.

Maybe it would be clearer to make the test body an async function and await the expectation rather than returning it? (see the second example in the code block at https://jestjs.io/docs/en/tutorial-async#async-await )

Reply:
I did follow one of the example on the same link you posted here (notice the comment on line 5 in this same file).
You may be right about clarity! I'll wait for more input from the team before deciding though.